File: backend/src/main/scala/microservices/traveler/objects/DeleteTravelerPlannerRequest.scala
// 本文件定义 traveler 模块的删除请求 `DeleteTravelerPlannerRequest`，仅负责数据结构与 JSON codec。
package com.typesafe.travel.traveler.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class DeleteTravelerPlannerRequest(
    actingUserId: String,
    ownerUserId: String,
    travelerId: String
)

object DeleteTravelerPlannerRequest:
  given sourceEncoder: Encoder[DeleteTravelerPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[DeleteTravelerPlannerRequest] = deriveDecoder

File: frontend/src/microservices/traveler/objects/DeleteTravelerPlannerRequest.ts
// 本文件定义 traveler 模块的 `DeleteTravelerPlannerRequest`，作为删除请求参数并提供 JSON 编解码。

export type DeleteTravelerPlannerRequest = {
  actingUserId: string
  ownerUserId: string
  travelerId: string
}

export const deleteTravelerPlannerRequestFromJson = (json: string): DeleteTravelerPlannerRequest =>
  JSON.parse(json) as DeleteTravelerPlannerRequest

export const deleteTravelerPlannerRequestToJson = (value: DeleteTravelerPlannerRequest): string =>
  JSON.stringify(value)
